### PR TITLE
Add check to submitConsent mutation and refactor

### PR DESCRIPTION
**Pull request by lucianak**
Adds a list of valid consents, and a check that a submitted consent is valid when the submitConsent mutation is called. Until now anything could be written into the consent policy field.

Additional refactoring: removed outdated 5-year campaign policy, added lots of tests.

### CODE REVIEW

**Comment by lucianak** [comment]
Todo:
- [ ] add translation for new error, does this need any updates in the frontend? 
- [ ] add tests to make sure deduplication works as expected (as we have quite a few duplicates in the DB)

File: packages/backend-modules/auth/graphql/resolvers/_mutations/submitConsent.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
-const { ensureSignedIn, Consents } = require('../../../lib')
```
Comment (lucianak) [inline packages/backend-modules/auth/graphql/resolvers/_mutations/submitConsent.js:1]: had to change this to make testing work

File: packages/backend-modules/auth/graphql/resolvers/_mutations/revokeConsent.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-const { ensureSignedIn, Consents } = require('../../../lib')
```
Comment (lucianak) [inline packages/backend-modules/auth/graphql/resolvers/_mutations/revokeConsent.js:1]: had to change this to make testing work

File: packages/backend-modules/auth/__tests__/lib/consents.u.jest.js [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+      ip: req.ip,
+    })
+    expect(saved).toContainEqual({
+      userId: userId,
+      policy: 'PROGRESS',
+      ip: req.ip,
+    })
+  })
+
+  test('saving fails', () => {
```
Comment (jstcki) [inline packages/backend-modules/auth/__tests__/lib/consents.u.jest.js:281]: Maybe describe why saving should fail in this test?

File: packages/backend-modules/auth/lib/Consents.js [hunk 4]
```diff
@@ -62,18 +58,16 @@ const lastRecordForPolicyForUser = async ({ userId, policy, pgdb }) =>
...
 
 const requiredConsents = async ({ userId, pgdb }) => {
-  const { ENFORCE_CONSENTS = '' } = process.env
-
   if (ENFORCE_CONSENTS) {
     const consented = userId ? await consentsOfUser({ userId, pgdb }) : []
 
-    return ENFORCE_CONSENTS.split(',').filter(
+    return ENFORCE_CONSENTS.filter(
       (consent) => consented.indexOf(consent) === -1,
```
Comment (jstcki) [inline packages/backend-modules/auth/lib/Consents.js:71]:
```suggestion
      (consent) => !consented.includes(consent),
```
Maybe a bit easier to read?

File: packages/backend-modules/auth/lib/Consents.js [hunk 5]
```diff
@@ -62,18 +58,16 @@ const lastRecordForPolicyForUser = async ({ userId, policy, pgdb }) =>
...
-  lastRecordForPolicyForUser(args).then(
+const statusForPolicyForUser = async ({ userId, policy, pgdb }) =>
+  lastRecordForPolicyForUser({ userId, policy, pgdb }).then(
     (record) => record && record.record === 'GRANT',
   )
 
 const requiredConsents = async ({ userId, pgdb }) => {
-  const { ENFORCE_CONSENTS = '' } = process.env
-
   if (ENFORCE_CONSENTS) {
```
Comment (jstcki) [inline packages/backend-modules/auth/lib/Consents.js:67]: This condition is not needed anymore?

**Review by hdahlheim — APPROVED** [review]
